lib/Default/bar.functions.inc

```diff
@@ -2,33 +2,35 @@
 
 function checkForLottoWinner($gameID) {
 	global $db;
-	//we we check for a lotto winner...
+	// we check for a lotto winner...
 	$db->lockTable('player_has_ticket');
 	$lottoInfo = getLottoInfo($gameID);
 	$lottoWon = false;
+
 	if ($lottoInfo['TimeRemaining'] <= 0) {
 		//we need to pick a winner
 		$db->query('SELECT * FROM player_has_ticket WHERE game_id = '.$db->escapeNumber($gameID).' AND time > 0 ORDER BY rand() LIMIT 1');
 		if ($db->nextRecord()) {
 			$winner_id = $db->getField('account_id');
 			$time = $db->getField('time');
 		}
+
+		// Any unclaimed prizes get merged into this prize
 		$db->query('SELECT * FROM player_has_ticket WHERE time = 0 AND game_id = '.$db->escapeNumber($gameID));
-		if ($db->nextRecord()) {
+		while ($db->nextRecord()) {
```

Comment:
You can optimise this by doing
```php
$db->query('SELECT SUM(prize) AS prize FROM player_has_ticket WHERE time = 0 AND game_id = '.$db->escapeNumber($gameID));
if ($db->nextRecord()) {
	$lottoInfo['Prize'] += $db->getField('prize');
}
```

Reply:
Thanks for the suggestion. I've made this change.